packages/api-server/src/models/user.ts

```diff
@@ -0,0 +1,134 @@
+import { QueryResult } from 'pg'
+
+import { UserModel } from '@portaler/types'
+
+import { dbQuery } from '../db'
+import { DiscordMe } from '../utils/discord/fetchUser'
+import { DiscordMeGuilds } from '../utils/discord/fetchUserGuilds'
+import logger from '../utils/logger'
+
+export type UserError = 'NoUser' | 'NoServers' | 'NoRoles'
+
+/**
+ * Gets a User from the database
+ * @param  discordId
+ * @returns UserModel
+ */
+export const getUser = async (
+  discordId: string
+): Promise<UserModel | UserError> => {
+  const {
+    rows: userRows,
+  } = await dbQuery('SELECT * FROM users WHERE discord_id = $1', [discordId])
+
+  if (userRows.length === 0) {
+    logger.info('Return')
+    return Promise.resolve('NoUser')
```

Comment:
This is an async function, can't you just return the string directly without Promise.resolve?

Reply:
Yeah I can. I just got into a groove. I wrote most of this without running it. It needs a lot of cleanup.